### PR TITLE
Refactor ec2_vpc_peer* modules

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Depends-On: https://github.com/ansible-collections/amazon.aws/pull/2303

Refactor ec2_vpc_peer* modules
Once the module is migrated to amazon.aws I will deprecate `result` returned by the info module and rename the ec2_vpc_peer module (see https://github.com/ansible-collections/community.aws/issues/2154). 

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME

ec2_vpc_peer
ec2_vpc_peering_info
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/20240924-ec2_vpc_peer-refactor.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - ec2_vpc_peer - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2``.
+  - ec2_vpc_peering_info - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2``.
```
Comment (abikouo) [inline changelogs/fragments/20240924-ec2_vpc_peer-refactor.yml:4]:
```suggestion
  - ec2_vpc_peer - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2`` (https://github.com/ansible-collections/community.aws/pull/2153).
  - ec2_vpc_peering_info - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2`` (https://github.com/ansible-collections/community.aws/pull/2153).
```

File: plugins/modules/ec2_vpc_peering_info.py [hunk 2]
```diff
@@ -191,41 +193,175 @@
...
     if module.params.get("peer_connection_ids"):
         params["VpcPeeringConnectionIds"] = module.params.get("peer_connection_ids")
+
     try:
-        result = client.describe_vpc_peering_connections(aws_retry=True, **params)
-        result = normalize_boto3_result(result)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        result = describe_vpc_peering_connections(client, **params)["VpcPeeringConnections"]
+    except AnsibleEC2Error as e:
         module.fail_json_aws(e, msg="Failed to describe peering connections")
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_peering_info.py:362]:
```suggestion
        module.fail_json_aws(e)
```

File: plugins/modules/ec2_vpc_peer.py [hunk 3]
```diff
@@ -335,237 +337,259 @@
...
-            "requester-vpc-info.vpc-id": params["PeerVpcId"],
-            "accepter-vpc-info.vpc-id": params["VpcId"],
-        }
-        result = client.describe_vpc_peering_connections(
-            aws_retry=True,
-            Filters=ansible_dict_to_boto3_filter_list(peer_filter),
+
+    try:
+        peering_connections = describe_vpc_peering_connections(
+            client, **{"Filters": ansible_dict_to_boto3_filter_list(filters)}
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_peer.py:413]:
```suggestion
            client, Filters=ansible_dict_to_boto3_filter_list(filters)
```

File: plugins/modules/ec2_vpc_peer.py [hunk 4]
```diff
@@ -335,237 +337,259 @@
...
         )
+        if peering_connections["VpcPeeringConnections"] == []:
+            # Try again with the VPC/Peer relationship reversed
+            filters = {
+                "requester-vpc-info.vpc-id": params["PeerVpcId"],
+                "accepter-vpc-info.vpc-id": params["VpcId"],
+            }
+            peering_connections = describe_vpc_peering_connections(
+                client, **{"Filters": ansible_dict_to_boto3_filter_list(filters)}
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_peer.py:422]:
```suggestion
                client, Filters=ansible_dict_to_boto3_filter_list(filters)
```

File: plugins/modules/ec2_vpc_peer.py [hunk 5]
```diff
@@ -335,237 +337,259 @@
...
+            )
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
 
-    return result
+    return peering_connections["VpcPeeringConnections"]
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_peer.py:427]:
```suggestion
    return peering_connections
```
can be updated according to https://github.com/ansible-collections/amazon.aws/pull/2303/files#r1778361791

File: plugins/modules/ec2_vpc_peer.py [hunk 6]
```diff
@@ -335,237 +337,259 @@
...
-    params = dict()
-    params["VpcPeeringConnectionIds"] = [peering_id]
+def get_peering_connection_by_id(client, module: AnsibleAWSModule, peering_id: str) -> Dict[str, Any]:
+    filters: Dict = {}
+    filters["VpcPeeringConnectionIds"] = [peering_id]
+
     try:
-        vpc_peering_connection = client.describe_vpc_peering_connections(aws_retry=True, **params)
-        return vpc_peering_connection["VpcPeeringConnections"][0]
+        result = describe_vpc_peering_connections(client, **filters)
```
Comment (abikouo) [inline plugins/modules/ec2_vpc_peer.py:533]:
```suggestion
        result = describe_vpc_peering_connections(client, VpcPeeringConnectionIds=[peering_id])
```

**Review by abikouo — COMMENTED** [review]
Just minor remarks

**Review by abikouo — COMMENTED** [review]
We should add support for `check_mode` for the `ec2_vpc_peer` module (the module is using the `ensure_ec2_tags` API which natively support the check mode).

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Review by abikouo — APPROVED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**